samples/packages/metadata-impact-report/src/main/kotlin/com/atlan/pkg/mdir/Reporter.kt

```diff
@@ -176,57 +176,73 @@ object Reporter {
         }
     }
 
-    private fun createGlossaryIdempotent(
+    private fun createDomainIdempotent(
         client: AtlanClient,
-        glossaryName: String,
-    ): Glossary =
+        domainName: String,
+    ): DataDomain =
         try {
-            Glossary.findByName(client, glossaryName)
+            DataDomain.findByName(client, domainName)[0]!!
         } catch (e: NotFoundException) {
-            val create =
-                Glossary
-                    .creator(glossaryName)
-                    .assetIcon(AtlanIcon.PROJECTOR_SCREEN_CHART)
-                    .build()
+            val create = DataDomain.creator(domainName).build()
             val response = create.save(client)
             response.getResult(create)
         }
 
-    private fun createCategoriesIdempotent(
+    private fun createSubDomainsIdempotent(
         client: AtlanClient,
-        glossary: Glossary?,
-    ): Map<String, String> {
-        if (glossary == null) return emptyMap()
+        domain: DataDomain?,
+    ): Pair<Map<String, String>, Map<String, String>> {
+        if (domain == null) return emptyMap<String, String>() to emptyMap<String, String>()
         val nameToResolved = mutableMapOf<String, String>()
-        val placeholderToName = mutableMapOf<String, String>()
+        val guidToResolved = mutableMapOf<String, String>()
+        val placeholderToName = mutableMapOf<DataDomain, String>()
+        val placeholderToGuid = mutableMapOf<String, String>()
         AssetBatch(client, 20).use { batch ->
-            CATEGORIES.forEach { (name, description) ->
+            SUBDOMAINS.forEach { (name, description) ->
                 val builder =
                     try {
-                        val found = GlossaryCategory.findByNameFast(client, name, glossary.qualifiedName)[0]
-                        found.trimToRequired().guid(found.guid)
+                        val subDomains = DataDomain.findByName(client, name)
+                        var found: DataDomain? = null
+                        for (sub in subDomains) {
+                            if (sub.parentDomain != null && sub.parentDomain.guid == domain.guid) {
+                                found = sub
+                                break
+                            }
+                        }
```

Comment:
Is it not possible to write our own query here along the lines of `DataDomain.select(client).where(DataDomain.PARENT_DOMAIN.eq(domain.qualifiedName)).where(DataDomain.NAME.eq(name))` ? This would avoid us having to retrieve many domains and then loop through them.

We should push-down the logic to pull back the bare minimum information we require, rather than risking pulling back more information than we require and paging through it all (multiple underlying API requests) through a loop.